### ISSUE TITLE
Regression with QuadGK@2.9

### ISSUE THREAD

**Issue by sathvikbhagavan** [JuliaMath/QuadGK.jl#89]
An issue was caught in https://github.com/SciML/DataInterpolations.jl/pull/178 where tests started failing with QuadGK@2.9

I have made a MWE to illustrate the issue (using a simple LinearInterpolation from DataInterpolations):

```julia
u = collect(0:10)
t = collect(0:10)
A = LinearInterpolation(u,t)
qint, err = quadgk(A, 0.0, 10.0)
```

It is a straight line with slope 1. So the integral should be 50.0.

![image](https://github.com/JuliaMath/QuadGK.jl/assets/35105271/e2198722-da6d-43a5-a131-77ab6ff977f7)

with QuadGK@2.8

```julia
julia> qint, err = quadgk(A, 0.0, 10.0)
(50.0, 0.0)
```

with QuadGK@2.9

```julia
julia> qint, err = quadgk(A, 0.0, 10.0)
(51.76911234248546, 4.274582013885939)
```

**Comment by stevengj** [comment]
cc @lxvm

**Comment by stevengj** [comment]
Seems to be a dispatch problem, since passing the argument as a function works:
```jl
julia> quadgk(A, 0.0, 10.0)
(51.76911234248546, 4.274582013885939)

julia> quadgk(t -> A(t), 0.0, 10.0)
(50.0, 0.0)
```

**Comment by stevengj** [comment]
Okay, the problem is that `LinearInterpolation` is a subtype of `AbstractVector`, so it is calling the wrong method of `QuadGK.evalrule`.

**Comment by stevengj** [comment]
Should be fixed by #90.  Will merge and tag when green.
